src/analysis/processing/qgsalgorithmremoveduplicatesbyattribute.cpp

```diff
@@ -0,0 +1,171 @@
+/***************************************************************************
+                         qgsalgorithmremoveduplicatesbyattribute.cpp
+                         ----------------------------------
+    begin                : October 2018
+    copyright            : (C) 2018 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmremoveduplicatesbyattribute.h"
+
+///@cond PRIVATE
+
+QString QgsRemoveDuplicatesByAttributeAlgorithm::name() const
+{
+  return QStringLiteral( "removeduplicatesbyattribute" );
+}
+
+QString QgsRemoveDuplicatesByAttributeAlgorithm::displayName() const
+{
+  return QObject::tr( "Remove duplicates by attribute" );
+}
+
+QStringList QgsRemoveDuplicatesByAttributeAlgorithm::tags() const
+{
+  return QObject::tr( "field,value,same,filter" ).split( ',' );
+}
+
+QString QgsRemoveDuplicatesByAttributeAlgorithm::group() const
+{
+  return QObject::tr( "Vector selection" );
+}
+
+QString QgsRemoveDuplicatesByAttributeAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorselection" );
+}
+
+void QgsRemoveDuplicatesByAttributeAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ),
+                QList< int >() << QgsProcessing::TypeVector ) );
+  addParameter( new QgsProcessingParameterField( QStringLiteral( "FIELDS" ), QObject::tr( "Field to match duplicates by" ), QVariant(), QStringLiteral( "INPUT" ), QgsProcessingParameterField::Any, true ) );
+
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr( "Filtered (no duplicates)" ) ) );
+  QgsProcessingParameterFeatureSink *failOutput = new QgsProcessingParameterFeatureSink( QStringLiteral( "DUPLICATES" ),  QObject::tr( "Filtered (duplicates)" ),
```

Comment:
Possible bonus (not blocking this PR in anyway): harmonize the remove duplicate geometries alg's capacity by adding a similar "DUPLICATES" output.

Reply:
Ok, I'll do that in a follow up work. I'd like to port that other alg to c++ anyway.